app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -186,11 +210,44 @@ func (s *Hijacker) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 	}
 }
 
-func (s *Hijacker) getStats(ctx context.Context, initReq *http.Request, app ApplicationToScrape) []byte {
+// selectContentType selects the highest priority content type supported by the applications.
+// If no valid content type is returned by the applications, it returns the highest priority
+// content type supported by the scraper.
+func selectContentType(contentTypes <-chan expfmt.Format, reqHeader http.Header) expfmt.Format {
+	// Tracks highest negotiated content type priority.
+	// Lower number means higher priority
+	//
+	// We should not simply use the highest priority content type even if `application/openmetrics-text`
+	// is the superset of `text/plain`, as it might not be
+	// supported by the applications or the user might be using older prom scraper.
+	// So it's better to choose the highest negotiated content type between the apps and the scraper.
+	var ctPriority int32 = 1<<31 - 1
```

Comment:
I did not use `math.MaxInt32` just to avoid an import. Initially I wrote `len(prometheusPriorityContentType)`, but it seemed bit weird, so used `1<<31 - 1` as initial value. 😅 
If you want I can update it anyway.